### DISCUSSION TITLE
Multiple microphones

### DISCUSSION THREAD

**Discussion by rulowa** [SaifAqqad/AHK_MicMute#18] [Ideas]
Hi, I have multiple microphones and from time to time I need to mute one off and unmute the other, can you add this support? TY!

**Comment by SaifAqqad** [comment]
Hey, I have some ideas about this but I need feedback,

So first off, I could just add hotkeys for profiles so you can easily switch between them and then you can mute/unmute different microphones on the fly.
Or I could also add the ability to turn on multiple profiles at the same time but that would require a rewrite for alot of things (e.g linked app), also it would require a check to make sure conflicting profiles are not turned on at the same time (e.g if two profiles control the same mic).

**Reply by rulowa** [reply]
I would like, of course, without switching profiles, do it as it is easier for you, thank for fast reply and great app!

**Reply by SaifAqqad** [reply]
> I would like, of course, without switching profiles, do it as it is easier for you, thank for fast reply and great app!
 
 Thanks, I think I have a good idea for this.
 Each profile can have a hotkey for any number of microphones you have, so the hotkeys are bound to each microphone instead of the profile, this allows you to control any microphones you have at the same time and still have different hotkeys for each profile

**Reply by SaifAqqad** [reply]
so the config file would look something like this: 
```json
{
    "DefaultProfile": "Default",
    "MuteOnStartup": 0,
    "Profiles": [
        {
            "afkTimeout": 0,
            "ExcludeFullscreen": 0,
            "LinkedApp": "",
            "Microphone": [
                {
                    "Name": "Microphone (AmazonBasics Desktop Mini Mic)",
                    "MuteHotkey": "~*RShift",
                    "UnmuteHotkey": "~*RShift",
                    "PushToTalk": 0
                },
                {
                    "Name": "Headset Microphone (Corsair HS70 Wireless Gaming Headset)",
                    "MuteHotkey": "*B",
                    "UnmuteHotkey": "*B",
                    "PushToTalk": 1
                }
            ],
            "OnscreenFeedback": 1,
            "OSDPos": {
                "x": 145,
                "y": 1000
            },
            "ProfileName": "Firefox",
            "PTTDelay": 100,
            "SoundFeedback": 1,
            "UpdateWithSystem": 1
        }
    ],
    "SwitchProfileOSD": 1,
    "UseCustomSounds": 0
}
```

**Reply by rulowa** [reply]
This is the perfect solution!

**Reply by SaifAqqad** [reply]
Hey, I just pushed a commit that added this feature
I rewrote alot of parts, so i'm currently testing it, and will be testing it for at least a week before making a release.
But if you want to test it, you can download the `latest_commit` version [here](https://github.com/SaifAqqad/AHK_MicMute/releases/tag/latest_commit), or you can just run `MicMute.ahk` directly if you have Autohotkey installed.
Any feedback on this feature or the whole update is welcome.

**Reply by rulowa** [reply]
Works great, thanks for your work.
